smithy4s-client/src/main/scala/kinesis4cats/smithy4s/client/KinesisClient.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023-2023 etspaceman
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kinesis4cats.smithy4s.client
+
+import cats.effect.syntax.all._
+import cats.effect.{Async, Resource}
+import cats.syntax.all._
+import com.amazonaws.kinesis._
+import org.http4s.client.Client
+import org.http4s.{Request, Response}
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import smithy4s.aws._
+import smithy4s.aws.http4s._
+
+import kinesis4cats.logging._
+import kinesis4cats.smithy4s.client.middleware.RequestResponseLogger
+
+object KinesisClient {
+
+  /** Helper class containing required
+    * [[kinesis4cats.logging.LogEncoder LogEncoders]] for the
+    * [[kinesis4cats.smithy4s.client.KinesisClient KinesisClient]]
+    */
+  final class LogEncoders[F[_]](implicit
+      val http4sRequestLogEncoder: LogEncoder[Request[F]],
+      val http4sResponseLogEncoder: LogEncoder[Response[F]]
+  )
+
+  def apply[F[_]](
+      client: Client[F],
+      region: AwsRegion,
+      creds: F[AwsCredentials]
+  )(implicit
+      F: Async[F],
+      LE: LogEncoders[F]
+  ): Resource[F, KinesisClient[F]] = for {
+    logger <- Slf4jLogger.create[F].toResource
```

Comment:
My two cents : I'm personally not a fan of slf4j being pulled by default. For starters it prevents you from cross-building to js/native down the road (though I don't think people will care much about js, native could be very interesting). 

Moreover, I just think libraries should never instantiate loggers ever. I think it's fine in this context to standardise against log4cats, but the logger should be injected as a parameter, just like the http client. 

Put differently : you're not choosing the http4s backend for your users, you shouldn't choose the logging backend for them ^^

Reply:
Fair enough. I'll see what I can do about abstracting the logging backend in a different PR.

Reply:
@Baccata Http4s uses the approach of having a default logger of Slf4jLogger, with the ability to override it on their builders. Would that be acceptable here for your concern?

Reply:
That's certainly a sensible approach

Reply:
@etspaceman http4s (ember, in particular) does that [on the JVM](https://github.com/http4s/http4s/blob/0e253e30bf983545cb0ebdd6407495b629f4ed1e/ember-server/jvm/src/main/scala/org/http4s/ember/server/EmberServerBuilderPlatform.scala#L25) but you'll see that the [JS/Native implementations](https://github.com/http4s/http4s/blob/0e253e30bf983545cb0ebdd6407495b629f4ed1e/ember-server/js-native/src/main/scala/org/http4s/ember/server/EmberServerBuilderPlatform.scala#L25) use a no-op logger by default. What is more, they're going to [get rid of it on the JVM too](https://github.com/http4s/http4s/pull/6645).

Reply:
I like the idea of using NoOp as a default and having some sort of builder where you can pass a different one.

Reply:
Alright. I've updated this to use a NoOp implementation by default.

I feel pretty safe leaving the SL4FJ dependency defaults in the java wrapper projects as they are provided by the underlying wrappers to begin with. But for this project, with it being native, it makes sense to have a choice here.